src/nvim/getchar.c

```diff
@@ -1743,21 +1741,27 @@ static int vgetorpeek(const int advance)
        * try re-mapping.
        */
       for (;; ) {
-        /*
-         * os_breakcheck() is slow, don't use it too often when
-         * inside a mapping.  But call it each time for typed
-         * characters.
-         */
-        if (typebuf.tb_maplen)
-          line_breakcheck();
-        else
-          os_breakcheck();                      /* check for CTRL-C */
         keylen = 0;
-        if (got_int) {
-          c = handle_int(advance);
-          exiting = true;
-          break;
-        } else if (typebuf.tb_len > 0) {
+
+        { // Check for CTRL-C
+          /*
+           * os_breakcheck() is slow, don't use it too often when
+           * inside a mapping.  But call it each time for typed
+           * characters.
+           */
+          if (typebuf.tb_maplen)
+            line_breakcheck();
+          else
+            os_breakcheck();
+
+          if (got_int) {
+            c = handle_int(advance);
+            exiting = true;
+            break;
+          }
+        }
+
+        if (typebuf.tb_len > 0) {
```

Comment:
In the `if` clause we `break` out of the `for(;;)` loop (moreover `exiting` is set to `true` so that the `while()` loop will exit too).
Whether the `else` keyword is used or not, the control flow can't go through both blocks.